code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -572,3 +609,109 @@
+	. = ..()
+	RegisterSignal(L, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(on_attack))
+	RegisterSignal(L, COMSIG_XENOMORPH_TAKING_DAMAGE, PROC_REF(on_take_damage))
+	ADD_TRAIT(L, TRAIT_BLOODTHIRSTER, TRAIT_GENERIC)
+	START_PROCESSING(SSprocessing, src)
+	last_fight_time = world.time
+
+/datum/action/ability/xeno_action/bloodthirst/remove_action(mob/living/L)
+	. = ..()
+	REMOVE_TRAIT(L, TRAIT_BLOODTHIRSTER, TRAIT_GENERIC)
+	UnregisterSignal(L, list(COMSIG_XENOMORPH_ATTACK_LIVING, COMSIG_XENOMORPH_TAKING_DAMAGE))
+	STOP_PROCESSING(SSprocessing, src)
+
+/datum/action/ability/xeno_action/bloodthirst/proc/on_attack(datum/source, mob/living/attacked, damage)
+	SIGNAL_HANDLER
+	if(!ishuman(attacked) || attacked.stat == DEAD)
+		return
+	last_fight_time = world.time
+
+/datum/action/ability/xeno_action/bloodthirst/proc/on_take_damage(datum/source, damage)
+	SIGNAL_HANDLER
+	last_fight_time = world.time
+
+/datum/action/ability/xeno_action/bloodthirst/process()
+	var/mob/living/carbon/xenomorph/xeno = owner
+	if(last_fight_time + decay_delay > world.time)
+		return
+	if(xeno.use_plasma(BLOODTHIRST_DECAY_PER_TICK))
+		disintegrating = FALSE
+		return
+	if(!disintegrating)
+		hit_zero_time = world.time
+		owner.balloon_alert(owner, "disintegrating...")
+		xeno.playsound_local(xeno, 'sound/voice/hiss5.ogg', 50)
+		disintegrating = TRUE
+		return
+	if((hit_zero_time + damage_delay) < world.time)
+		//take  damage per tick down to a minimum allowed hp
+		var/damage_taken = min(MAX_DAMAGE_PER_DISINTEGRATING, (xeno.health - xeno.health_threshold_crit - LOWEST_BLOODTHIRST_HP_ALLOWED))
+		xeno.take_overall_damage(damage_taken)
+
+
+#define DEATHMARK_DAMAGE_OR_DIE 400
+#define DEATHMARK_DURATION 40 SECONDS
+#define DEATHMARK_MESSAGE_COOLDOWN 2 SECONDS
+
+/datum/action/ability/xeno_action/deathmark
+	name = "deathmark"
+	desc = "Mark yourself for death, filling your bloodthirst, but failing to deal enough damage to living creatures while it is active instantly kills you."
+	cooldown_duration = DEATHMARK_DURATION*3
+	COOLDOWN_DECLARE(message_cooldown)
+	var/damage_dealt = 0
+
+/datum/action/ability/xeno_action/deathmark/action_activate()
+	var/mob/living/carbon/xenomorph/xeno = owner
+	addtimer(CALLBACK(src, PROC_REF(on_deathmark_expire)), DEATHMARK_DURATION)
+
+	xeno.overlays_standing[SUIT_LAYER] = image('icons/Xeno/64x64_Xeno_overlays.dmi', icon_state = "deathmark")
+	xeno.apply_temp_overlay(SUIT_LAYER, DEATHMARK_DURATION)
+
+	RegisterSignal(owner, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(on_attack))
+	damage_dealt = 0
+	xeno.use_plasma(-xeno.xeno_caste.plasma_max) // fill it to the max so they can kill better
+	xeno.add_movespeed_modifier(MOVESPEED_ID_RAVAGER_DEATHMARK, TRUE, 0, NONE, TRUE, -1.5) //Extra speed so they can get to where to kill better
+	xeno.emote("roar")
+
+/// on attack for deathmark, tracks the amount of dmg dealt
+/datum/action/ability/xeno_action/deathmark/proc/on_attack(datum/source, mob/living/attacked, damage)
```

Comment:
So, assuming that the rav only really slashes to try to fulfill deathmark's 400 damage threshold, as COMSIG_XENOMORPH_ATTACK_LIVING outputs xeno's slash damage times whatever damage modifier it has, but I'll just assume it's 25 as it usually is, 400/25, means you need to do 16 slashes to not die, which means you'd probably have to kill two marines just to not die, unless they're incredibly tanky because of tyr 2

Reply:
Means you have 2.5 seconds to do each slash or you'll simply run out of time.

Reply:
Just use the screen timer thingy or do a similar thing on the mob itself